File: 0128-longest-consecutive-sequence/0128-longest-consecutive-sequence.java
class Solution {
    public int longestConsecutive(int[] nums) {
        HashSet<Integer> set = new HashSet<>();
        int res = 0;
        
        for(int num : nums){
            set.add(num);
        }
        
        for(int num : nums){
            int cnt = 1;
            if(!set.contains(num-1)){
                while(set.contains(num+1)){
                    cnt++;
                    num++;
                }
            }
            res = Math.max(res, cnt);
        }
        
        return res;
    }
}

File: 0128-longest-consecutive-sequence/0128-longest-consecutive-sequence.py
class Solution:
    def longestConsecutive(self, nums: List[int]) -> int:
        longest = 0
        nums_set = set(nums)
        
        for num in nums_set:
            if num - 1 not in nums_set:
                curr = num
                curr_len = 1
                
                while curr + 1 in nums_set:
                    curr += 1
                    curr_len += 1
                
                longest = max(longest, curr_len)
        return longest
        